### DISCUSSION TITLE
Custom useFetch key issue

### DISCUSSION THREAD

**Discussion by aligzl** [nuxt/nuxt#17261] [Questions] [answered]
Hello,

please check useApi composable, if i remove provide key on options useFetch sets same data to values on multi fetch but everything works notmal when i set a key.

[Reproduction](https://stackblitz.com/edit/github-cdqcsb?file=app.vue)

useApi  Composable
```
const useApi = async (url, options, useToken = true) => {
  const accessToken = useCookie("sid")
  const headers = useToken ? { ...(options?.headers || {}), ...(accessToken.value && { Authorization: `Bearer ${accessToken.value}` }) } : options?.headers || {}
  return useFetch(url, {
    ...options,
    baseURL: process.env.NODE_ENV === 'development'? "http://127.0.0.1:4000" : "https://someapi.com",
    initialCache: false,
    headers,
    key: options?.key || url.toString()
  })
}
 
export default useApi
```

**Comment by danielroe** [comment, marked as answer]
Setting a key is now required in this use case since https://github.com/nuxt/framework/pull/4955.

I plan to update the documentation to make this clearer.

**Reply by aligzl** [reply]
Thanks for your reply, it's good to know
